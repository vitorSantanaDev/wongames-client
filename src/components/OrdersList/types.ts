import { GameItemProps } from 'components/GameItem/types'

export type OrdersListProps = {
  items?: GameItemProps[]
}
