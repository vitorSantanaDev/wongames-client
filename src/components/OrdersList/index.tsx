import Heading from 'components/Heading'
import GameItem from 'components/GameItem'

import { OrdersListProps } from './types'

import * as S from './styles'
import Empty from 'components/Empty'

const OrdersList = ({ items }: OrdersListProps) => (
  <S.Wrapper>
    <Heading lineBottom color="black" size="small">
      My orders
    </Heading>
    {items?.length ? (
      items.map((item, index) => (
        <GameItem key={`${item.title}-${index}`} {...item} />
      ))
    ) : (
      <Empty
        title="You have no orders yet"
        description="Go back to the store and explore great games and offers"
      />
    )}
  </S.Wrapper>
)

export default OrdersList
