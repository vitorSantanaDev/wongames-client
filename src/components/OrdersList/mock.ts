import { GameItemProps } from 'components/GameItem/types'

export default [
  {
    image:
      'https://images.hdqwalls.com/download/cyberpunk-biker-gang-4k-cf-1360x768.jpg',
    title: 'Red Dead Redemption 2',
    price: 'R$ 215,00',
    downloadLink: 'https://wongames.com/game/download/yuYT56Tgh431LkjhNBgdf',
    paymentInfo: {
      flag: 'mastercard',
      image: '/img/Master.png',
      number: '*** *** **** 4326',
      purchasedDate: 'Purchase made on 07/20/2020 at 20:32'
    }
  },
  {
    image:
      'https://images.hdqwalls.com/download/cyberpunk-biker-gang-4k-cf-1360x768.jpg',
    title: 'Red Dead Redemption 2',
    price: 'R$ 215,00',
    downloadLink: 'https://wongames.com/game/download/kjhejl867asd76DEh',
    paymentInfo: {
      flag: 'visa',
      image: '/img/visa.png',
      number: '*** *** **** 5423',
      purchasedDate: 'Purchase made on 08/20/2020 at 21:12'
    }
  }
] as GameItemProps[]
