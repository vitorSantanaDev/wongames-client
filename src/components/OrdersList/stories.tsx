import { Story, Meta } from '@storybook/react'

import OrdersList from '.'

import { OrdersListProps } from './types'

import ordersMock from './mock'

export default {
  title: 'Profile/OrdersList',
  component: OrdersList,
  args: { items: [...ordersMock] }
} as Meta

export const Default: Story<OrdersListProps> = (args) => (
  <div style={{ maxWidth: 850, margin: 'auto' }}>
    <OrdersList {...args} />
  </div>
)
