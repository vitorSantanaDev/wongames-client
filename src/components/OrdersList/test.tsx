import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import OrdersList from '.'

import ordersMock from './mock'

jest.mock('components/Empty', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="Mock Empty" />
  }
}))

jest.mock('components/GameItem', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock GameItem">{children}</div>
  }
}))

describe('<OrdersList />', () => {
  it('should render the game items', () => {
    const { container } = renderWithTheme(
      <OrdersList items={[...ordersMock]} />
    )

    const heading = screen.getByRole('heading', { name: /my orders/i })
    expect(heading).toBeInTheDocument()

    const gameItems = screen.getAllByTestId('Mock GameItem')
    expect(gameItems).toHaveLength(2)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render empty state', () => {
    renderWithTheme(<OrdersList items={[]} />)

    const gameItems = screen.queryAllByTestId('Mock GameItem')
    expect(gameItems).toHaveLength(0)

    const emptyComponent = screen.getByTestId('Mock Empty')
    expect(emptyComponent).toBeInTheDocument()
  })
})
