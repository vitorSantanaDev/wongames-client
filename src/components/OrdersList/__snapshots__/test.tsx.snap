// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersList /> should render the game items 1`] = `
.c0 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c0::after {
  content: '';
  left: 0;
  bottom: -1rem;
  width: 5rem;
  position: absolute;
  border-bottom: 0.5rem solid #F231A5;
}

.c0::after {
  width: 3rem;
}

<main
  class=""
>
  <h2
    class="c0"
    color="black"
  >
    My orders
  </h2>
  <div
    data-testid="Mock GameItem"
  />
  <div
    data-testid="Mock GameItem"
  />
</main>
`
